sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -688,6 +690,10 @@ export class ApplicationStudentsController extends BaseController {
     );
     const [firstDisbursement, secondDisbursement] = disbursements;
     const [scholasticStandingChange] = application.studentScholasticStandings;
+
+    const ecertFailedValidations =
```

Comment:
Can it be moved to be in parallel alongside the already in place `await Promise.all`?

Reply:
The `await Promise.all` in line 672 is inside the if condition for Completed applications. This `ecertFailedValidations` checks validation for pending applications, so it won't be accessed if we move inside the if condition.

Reply:
I am having a hard time trying to find the `if` condition. The entire method `getCompletedApplicationDetails` is dedicated to dealing with a completed application, what am I missing?

Reply:
Oh. I got mixed up with application status and disbursement schedule status. The application status is completed already, and we are checking the disbursements with pending status. I will add it to the `Promise all` in line 672.